File: backend/StorageModule.cpp
#include "StorageModule.h"

StorageModule::StorageModule() {
    products = {
        {1, "Laptop", 1200.0, 10, "Electronics"},
        {2, "Mouse", 25.0, 50, "Electronics"},
        {3, "Keyboard", 45.0, 30, "Electronics"},
        {4, "Coffee Mug", 12.0, 100, "Lifestyle"}
    };
}

std::vector<Product> StorageModule::getProducts() const {
    return products;
}

void StorageModule::addProduct(const Product& product) {
    products.push_back(product);
}

Product* StorageModule::getProductById(int id) {
    for (auto& p : products) {
        if (p.id == id) return &p;
    }
    return nullptr;
}

File: src/modules/StorageModule.js
/**
 * StorageModule.js
 * Responsibility: Handle persistent storage of product data.
 */
export const StorageModule = {
  products: [
    { id: 1, name: 'Laptop', price: 1200, category: 'Electronics' },
    { id: 2, name: 'Mouse', price: 25, category: 'Electronics' },
    { id: 3, name: 'Keyboard', price: 45, category: 'Electronics' },
    { id: 4, name: 'Coffee Mug', price: 12, category: 'Lifestyle' }
  ],

  getProducts() {
    return this.products;
  },

  addProduct(product) {
    const newProduct = { ...product, id: Date.now() };
    this.products.push(newProduct);
    return newProduct;
  }
};
